python/cugraph/components/cc_wrapper.pyx

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c_cc cimport *
+#from libcpp cimport bool
+from c_graph cimport *
+#from libcpp cimport cugraph_cc_t
+from libc.stdint cimport uintptr_t
+import cudf
+#from pygdf import Column
+import numpy as np
+
+cpdef weak_cc(G, connect_type = CUGRAPH_WEAK):
+    """
+    Find the distances and predecessors for a breadth first traversal of a graph.
+    
+    Parameters
+    ----------
+    G : cugraph.graph
+        cuGraph graph descriptor, should contain the connectivity information as an
```

Comment:
This should accept edge list too.

Reply:
Are you saying the comment should reflect that? Not clear what you meant.

Reply:
The doc should say :

```
G : cugraph.graph
cuGraph graph descriptor, should contain the connectivity information as an edge list (edge weights are not used for this algorithm). The adjacency list will be computed if not already present.
```

The feature should be updated if it is not supported.

Reply:
Afaik edge list is the expected input in Rapids pipelines in python.

@BradReesWork please confirm if you want the python API of this analytics to support only CSR or if we should add an edge list to adjacency list conversion when the user only has an edge list.

Reply:
I thought that cugraph.Graph accepted either and created the CSR from there.  The C/CUDA code should then be based on a CSR (C/CUDA code does not need top match Nx)

Reply:
@afender, can you point me to a python wrapper example that does exactly what you want here?

Reply:
Any existing analytics wrappers should work as an example. 
You need to generate the adj list when entering your python function (it will be skipped if it already exists), that should go in line 53 I believe.

```
    err = gdf_add_adj_list(<gdf_graph*> graph)
    cudf.bindings.cudf_cpp.check_gdf_error(err)
```

Reply:
Okay, added the adj_list call above.